### DISCUSSION TITLE
Hidden value for API Protection Rule

### DISCUSSION THREAD

**Discussion by ralacerda** [pocketbase/pocketbase#5340] [Q&A] [answered]
Hello everyone,

I'm trying to create an API Protection Rule for one of my collections.
I want to check if the authorization header has an secret only my backend functions will have access to.
Something like:

```
@request.headers.x_password = "super_secret"
```

My problem is that this will be shown on the migration file.

```
collection.listRule = "@request.headers.x_password = \"super_secret\""
```

Is there a way to keep this hidden, maybe using an `.env` file?

**Comment by ganigeorgiev** [comment, marked as answer]
It is not possible. API rules are part of the collection definition and they don't allow resolving values from `os.Environ` or other global variables unless it is a predefined macro or db column value (_all API rules are resolved to plain SQL statements; you can inspect them with the `--dev` flag_).

One eventual alternative could be to create a helper collection (e.g. `secrets`), add your secret as a record and then use the `@collection.*` filter to check if such record exists:

```js
// assuming that there is at least one "secrets" record
@request.headers.x_password ?= @collection.secrets.secret
```

_Side-note: keep in mind that comparing plain secrets/hashes directly is vulnerable to timing attacks and it is recommend to use a different approach if you can avoid it._
